### ISSUE TITLE
Question: How to deal with static equals?

### ISSUE THREAD

**Issue by spirritus** [jqno/equalsverifier#417]
Hi there, 

found your project yesterday, I am amazed. Maybe I'm doing something wrong, I have a class with several static equals, how to handle these? For example

```
	/**
	 * Test for the floating-point equality between Complex objects.
	 * It returns {@code true} if both arguments are equal or within the
	 * range of allowed error (inclusive).
	 *
	 * @param x First value (cannot be {@code null}).
	 * @param y Second value (cannot be {@code null}).
	 * @param maxUlps {@code (maxUlps - 1)} is the number of floating point
	 *            values between the real (resp. imaginary) parts of {@code x} and
	 *            {@code y}.
	 * @return {@code true} if there are fewer than {@code maxUlps} floating
	 *         point values between the real (resp. imaginary) parts of {@code x}
	 *         and {@code y}.
	 * @see Precision#equals(double,double,int)
	 */
	public static boolean equals(Complex x, Complex y, int maxUlps) {
		return Precision.equals(x.real, y.real, maxUlps) && Precision.equals(x.imaginary, y.imaginary, maxUlps);
	}
```

Cheers

**Comment by jqno** [comment]
Hi!

I'm glad you like it!

I have to disappoint you though. EqualsVerifier is intended only to test the `equals` method that is inherited from `java.lang.Object`: https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Object.html#equals(java.lang.Object)

The methods in your class are also called `equals`, but they're not "the" `equals` method of an object. They don't override the existing `equals` method in `Object`. So unfortunately, you'll have to test these methods yourself.

**Comment by spirritus** [comment]
Thank you for the answer. Awesome, then how to tell EqualsVerifier to ignore those? Currently it complains about:
```
java.lang.AssertionError: EqualsVerifier found a problem in class lib.Complex.
-> Overloaded: More than one equals method found.
```

**Comment by jqno** [comment]
Well, I guess you can't. The reason why it throws this error, is that usually it's a mistake to define an `equals` method that isn't an override of `Object`. But that applies to non-static methods.

Do I understand correctly that your class contains both values which warrant an override of `Object.equals`, but also one or more of these utility equals methods?

**Comment by spirritus** [comment]
The idea is to represent a complex number by "wrapping" two doubles. The normal equals override deals with the ususal stuff.
The static methods are for equality in terms of an epsilon range, basically assume two complex numbers c1 and c2 are equal if their differenence is smaller than an epsilon value.

Maybe include an option in EqualsVerifier to ignore static equal methods, or throw only a warning instead of an error if EqualsVerifier encounters such methods. As you said, static equals are not part of the equals contract.

**Comment by jqno** [comment]
I agree, EqualsVerifier can ignore static `equals` methods. The check was mainly intended for cases where people write an overloaded `public boolean equals(MyClass x)` instead of having `public boolean equals(Object x)`.

**Comment by jqno** [comment]
I've just released version 3.6 for you!

**Comment by spirritus** [comment]
most awesome!
